### PR TITLE
Add goup handling in JSON transformations.

### LINKED ISSUE #49: targetOfjs does not convert groups
## 🐛 Bug Report

Function targetOfjs does not convert groups.

## To Reproduce

Using your sample Xliff 2.0 data:
```js
const file = path.resolve(__dirname, './sample.xlf');
const xliff = fs.readFileSync(file).toString();
const js = await xliff2js(xliff);
const target = targetOfjs(js);
console.log(target);
```
The output is:
```js
{
  key1: 'Hallo',
  key2: 'Eine Applikation um XLIFF Dokumente zu manipulieren und verarbeiten',
  'key.nested': 'XLIFF Daten Manager',
  group: undefined
}
```

## Expected behavior

I would expect the following output:
```js
{
  key1: 'Hallo',
  key2: 'Eine Applikation um XLIFF Dokumente zu manipulieren und verarbeiten',
  'key.nested': 'XLIFF Daten Manager',
  group: {
    groupUnit: 'Gruppe'
  }
}
```

## Your Environment

- *runtime version*: node v16.13.0
- *module version*: >=5.7.2
- *os*: Windows

### PR DESCRIPTION
**Pull request by marmarosi**
Resolves #49. Usage:
```
const res = targetOfjs(js, {includeGroups: true})
const res = sourceOfjs(js, {includeGroups: true})
```

### CODE REVIEW

**Comment by adrai** [comment]
Can you include a test?

**Comment by adrai** [comment]
Do you also want to support groups in the createxliff functions? If so, should we skip the includeGroups option completely and respect groups always (as soon as the object property is an object and not a string?)
What do you think?

**Comment by marmarosi** [comment]
It seems feasible. I give it a try...

**Comment by adrai** [comment]
Can you write the tests like the existing ones?

**Comment by marmarosi** [comment]
1. sourceOfjs & targetOfjs:
	options { includeGroups: true } removed
	handle nested groups as well
	handle group notes
2, js2xliff & js2xliff12:
	handle groups and nested groups
	handle group notes
3. Bugfix: js2xliff & js2xliff12 convert multiple notes correctly
	instead of <note>note1,note2</note>
	now <note>note1</note><note>note2</note>

I cannot run browserify and mocha tests on Windows, therefore created some functional test in test/mj/*.

**Comment by marmarosi** [comment]
I get the following errors:

$ ./node_modules/.bin/browserify --standalone xliff cjs/index.js -o xliff.js

D:\Development\logikum\xliff\cjs\index.js:1
import xliff2js from './xliff2js.js';
^
ParseError: 'import' and 'export' may appear only with 'sourceType: module'

$ ./node_modules/.bin/mocha test -R spec

D:\Development\logikum\xliff\cjs\index.js:1
import xliff2js from './xliff2js.js';
^^^^^^

SyntaxError: Cannot use import statement outside a module

**Comment by adrai** [comment]
Tests should run also on windows... what's the problem?

**Comment by adrai** [comment]
Which node version are you using?

**Comment by marmarosi** [comment]
node.js v16.3

**Comment by marmarosi** [comment]
I have found this idea: https://stackoverflow.com/questions/40029113/syntaxerror-import-and-export-may-appear-only-with-sourcetype-module-g, but do not know how to apply.

**Comment by adrai** [comment]
Ok, it should work on windows now: https://github.com/locize/xliff/runs/4684040716?check_suite_focus=true
Can you update your branch?

**Comment by marmarosi** [comment]
I check it...

**Comment by marmarosi** [comment]
Funny... I see your changes on Github, but git does not fetch anything.

**Comment by adrai** [comment]
https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork

File: lib/createjs.js [hunk 1]
```diff
@@ -34,14 +52,7 @@ const createjsClb = (srcLng, trgLng, srcKeys, trgKeys, ntKeys, ns, cb) => {
...
     Object.keys(srcKeys[ns]).forEach((srcKey) => {
-      js.resources[ns][srcKey] = {
-        source: srcKeys[ns][srcKey] || '',
-        target: trgKeys[ns][srcKey] || ''
-      }
-
-      if (ntKeys && ntKeys[ns] && ntKeys[ns][srcKey]) {
-        js.resources[ns][srcKey].note = ntKeys[ns][srcKey]
-      }
+      setSegment(js.resources[ns], srcKeys, trgKeys, ntKeys, srcKey)
```
Comment (adrai) [inline lib/createjs.js:55]: this should probably be something like: `setSegment(js.resources[ns], srcKeys[ns], trgKeys[ns], ntKeys && ntKeys[ns], srcKey)`

File: README.md [hunk 2]
```diff
@@ -118,6 +118,16 @@ const res = targetOfjs(js)
 //   "key2": "Eine Applikation um XLIFF Dokumente zu manipulieren und verarbeiten",
 //   "key.nested": "XLIFF Daten Manager"
 // }
+const res = targetOfjs(js, {includeGroups: true})
```
Comment (adrai) [inline README.md:121]: not needed anymore

**Comment by adrai** [comment]
> https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork

Are you able to sync your fork?

**Comment by marmarosi** [comment]
Still cannot run tests. I did successfully:
git fetch upstream
git checkout master
git merge upstream/master
...and I get the same errors. Even when I change to include-groups branch.

**Comment by adrai** [comment]
From what I see published here (https://github.com/marmarosi/xliff/tree/include-groups), I can't see any updates?

And based on the Github Windows tests, it works successfully: https://github.com/locize/xliff/runs/4684040716?check_suite_focus=true

**Comment by adrai** [comment]
If you prefer, you can delete everything and create a new PR...

File: README.md (hunk 2 above)
Comment (marmarosi) [reply README.md:121]: I agree, I correct it.

**Comment by marmarosi** [comment]
Adding ,/node_modules/.bin to PATH resolved my babel problem. However, some tests now fails, I suppose as a result of my amendments. I need to review them...

File: lib/createjs.js (hunk 1 above)
Comment (adrai) [reply lib/createjs.js:55]: @marmarosi this is for sure one of the test fails

Comment (marmarosi) [reply lib/createjs.js:55]: You're right, all tests succeeded having applied your patch. I just need to add some new tests for the changes.

**Comment by marmarosi** [comment]
I think the PR is error free now.

**Comment by adrai** [comment]
thank you for your contribution
released with v6.0.0

**Pull request merged**